Add UnitTest

HKhangVu opened the pull request.

HKhangVu committed:
Add UnitTest for reducers and render of  Some screens
Refactor:
+ Add test file in reducer folder
+ Add test file for App.tsx
+ Add test file for Home screen
+ Add test file for Not Found screen
+ Add test file for Root.tsx
+ Add test file for Navigation

khacanh commented on src/components/AppliedRoute/index.test.tsx:
```diff
@@ -0,0 +1,29 @@
+import Navigation from '../Navigation'; 
+
+describe("test render AppliedRoute", () => {
+    
```
No new line, also wrong indent

khacanh commented on src/components/AppliedRoute/index.test.tsx:
```diff
@@ -0,0 +1,29 @@
+        const wrapper = shallow(<AppliedRoute component = { Home } props = {"string"} {...rest}  />);
+        expect(wrapper).toMatchSnapshot();
+    });
+    test('renders AppliedRoute do not have props', () => { 
```
New line before a new test case

khacanh commented on src/components/AppliedRoute/index.test.tsx:
```diff
@@ -0,0 +1,29 @@
+describe("test render AppliedRoute", () => {
+    
+    test('renders AppliedRoute have all params', () => { 
+        let rest : { [index:string] :  string } = {}
```
No space in `index:string`, and double spaces in `] :   string`

khacanh commented on src/screens/NotFound/index.test.tsx:
```diff
@@ -0,0 +1,8 @@
+import NotFound from './index';
+import React from 'react';
+import { shallow } from 'enzyme';
```
3rd party imports before local imports

khacanh commented on src/screens/NoteDetail/index.test.tsx:
```diff
@@ -0,0 +1,13 @@
+import NoteDetail from './index';
+
+test('renders note detail', () => {
+    // const wrapper = shallow(
```
No need to keep commented out code

HKhangVu committed:
Code Reviewer: Update styling and add full test for
AppliedRoute  and AuthenticatedRoute
Refactor:
- Fix Styling for NoteDetail  NotFound
- Fix Styling and add more test for AppliedRouted
- Add test for AuthenticatedRoute

HKhangVu committed:
Add new line to the end of file

HKhangVu committed:
Add Render Unit Test to Screens and Components
also fix some styling.
Refactor:
+ Modify styling for App.test.tsx
+ Modify styling for AuthenticatedRoute test file
+ Modify styling for Signup test file
+ Add test render for Navigation
+ Add test render for UnauthenticatedRoute
+ Add test render for Login
+ Add test render for NewNote

HKhangVu committed:
update styling

HKhangVu committed:
Update indent space

HKhangVu committed:
fix indent

leo-le-07 commented on src/components/AppliedRoute/index.test.tsx:
```diff
@@ -0,0 +1,40 @@
+import store from '../../store';
+import AppliedRoute from './index';
+
+describe("test render AppliedRoute", () => {
```
`test` keyword is the most unnecessary in your test 😄 I see it everywhere. We're already in test file, everything we do here is for test, so no need that keyword anymore.
The test structure should be:
```
describe(<Object you wanna test>)
...test(<scenario 1>)
...test(<scenario 2>)
...test(<scenario 3>)
```

HKhangVu replied:
Thank you for your code review, i have changed the test name as you suggest in the new commit also the indentation of all files

HKhangVu committed:
fix indent

HKhangVu committed:
fix indentation

HKhangVu committed:
fix indetation

HKhangVu committed:
Add test for log out button and siteId on netlify
Refactor:
- Modify Navigation test file
- Add id site of netlify

HKhangVu committed:
Add 100% test for Navigation and UnauthenticatedRoute
Refactor:
- Add test in index.test.tsx
- Modify UnauthenticatedRoute index file to test function querystring
- Modify Navigation index to remove console command
